src/js/components/Select/index.d.ts

```diff
@@ -51,8 +64,9 @@ export interface SelectProps {
   valueKey?:
     | string
     | { key: string; reduce?: boolean }
-    | ((...args: any[]) => any);
+    | ((option: { label: string; value: number }) => string);
```

Comment:
we might still need the `any` option @jcfilben can you please confirm?

Reply:
Yes, I think the way it is setup right now the option is too restrictive. According to documentation: "If a function is provided, it is called with the option and the return value indicates the value. If reduce is true, the value coming via the key will be used for the onChange value and the value property is expected to be reduced to align."